src/minidump.rs

```diff
@@ -1188,12 +1195,158 @@ impl<'a> MinidumpStream<'a> for MinidumpSystemInfo {
+                    (md::ArmElfHwCaps::HWCAP_THUMB, "thumb"),
+                    (md::ArmElfHwCaps::HWCAP_26BIT, "26bit"),
+                    (md::ArmElfHwCaps::HWCAP_FAST_MULT, "fastmult"),
+                    (md::ArmElfHwCaps::HWCAP_FPA, "fpa"),
+                    (md::ArmElfHwCaps::HWCAP_VFP, "vfpv2"),
+                    (md::ArmElfHwCaps::HWCAP_EDSP, "edsp"),
+                    (md::ArmElfHwCaps::HWCAP_JAVA, "java"),
+                    (md::ArmElfHwCaps::HWCAP_IWMMXT, "iwmmxt"),
+                    (md::ArmElfHwCaps::HWCAP_CRUNCH, "crunch"),
+                    (md::ArmElfHwCaps::HWCAP_THUMBEE, "thumbee"),
+                    (md::ArmElfHwCaps::HWCAP_NEON, "neon"),
+                    (md::ArmElfHwCaps::HWCAP_VFPv3, "vfpv3"),
+                    (md::ArmElfHwCaps::HWCAP_VFPv3D16, "vfpv3d16"),
+                    (md::ArmElfHwCaps::HWCAP_TLS, "tls"),
+                    (md::ArmElfHwCaps::HWCAP_VFPv4, "vfpv4"),
+                    (md::ArmElfHwCaps::HWCAP_IDIVA, "idiva"),
+                    (md::ArmElfHwCaps::HWCAP_IDIVT, "idivt"),
+                ];
+
+                let mut cpu_info = format!("ARMv{}", raw.processor_level);
+
+                // Try to extract out known vendor/part names from the cpuid,
+                // falling back to just reporting the raw value.
+                let cpuid = arm_info.cpuid;
+                if cpuid != 0 {
+                    let vendor_id = (cpuid >> 24) & 0xff;
+                    let part_id = cpuid & 0xff00fff0;
+
+                    if let Some(&(_, vendor)) = vendors.iter().find(|&&(id, _)| id == vendor_id) {
+                        write!(&mut cpu_info, " {}", vendor).unwrap();
+                    } else {
+                        write!(&mut cpu_info, " vendor(0x{:x})", vendor_id).unwrap();
+                    }
+
+                    if let Some(&(_, part)) = parts.iter().find(|&&(id, _)| id == part_id) {
+                        write!(&mut cpu_info, " {}", part).unwrap();
+                    } else {
+                        write!(&mut cpu_info, " part(0x{:x})", part_id).unwrap();
+                    }
+                }
+
+                // Report all the known hardware features.
+                let elf_hwcaps = md::ArmElfHwCaps::from_bits_truncate(arm_info.elf_hwcaps);
+                if !elf_hwcaps.is_empty() {
+                    cpu_info.push_str(" features: ");
+
+                    // Iterator::intersperse is still unstable, so do it manually
+                    let mut comma = "";
+                    for &(_, feature) in features
+                        .iter()
+                        .filter(|&&(feature, _)| elf_hwcaps.contains(feature))
+                    {
+                        cpu_info.push_str(comma);
+                        cpu_info.push_str(feature);
+                        comma = ",";
+                    }
+                }
+
+                Some(cpu_info)
+            }
+            _ => None,
+        };
```

Comment:
Drive-by comment. This is the kind of function that could use some splitting for the sake of readability (and to cut down on heavy indentation). You could split functionality out into smaller more specific functions so for example this could be:
```
let cpu_info = match cpu {
  Cpu::X86 | Cpu::X86_64 = get_x86_cpu_info(...),
  Cpu::Arm = get_arm_cpu_info(...),
}
```
etc...